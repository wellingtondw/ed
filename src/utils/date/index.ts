type GetFullYearParams = {
  date: string | Date;
};
export const getFullYear = ({ date }: GetFullYearParams) => {
  return new Date(date).getFullYear();
};
