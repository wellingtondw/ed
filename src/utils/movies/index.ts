import { MovieCardListDTO } from 'components/MovieCardList';

export type MoviesDTO = {
  poster_path: string;
  title: string;
  release_date: string;
  vote_average: number;
  id: number;
};

type FormatMoviesListParams = {
  items: MoviesDTO[];
};

export const formatMoviesList = ({ items }: FormatMoviesListParams) => {
  return items.map((item): MovieCardListDTO => {
    const { poster_path, title, release_date, vote_average, id } = item;

    return {
      posterImage: poster_path,
      title,
      date: release_date,
      rating: vote_average,
      id
    };
  });
};
