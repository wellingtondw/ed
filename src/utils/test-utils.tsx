import { createStore } from 'redux';
import { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { createMemoryHistory } from 'history';

import { ThemeProvider } from 'styled-components';
import theme from 'styles/theme';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import rootReducer from '../store/modules/rootReducer';

const memoryHistory = createMemoryHistory();

const customRender = (
  ui: ReactElement,
  initialState = {},
  store = createStore(rootReducer, initialState),
  history = memoryHistory,
  { ...renderOptions }: RenderOptions = {}
) =>
  render(
    <Provider store={store}>
      <Router navigator={history} location={history.location}>
        <ThemeProvider theme={theme}>{ui}</ThemeProvider>
      </Router>
    </Provider>,
    renderOptions
  );

export * from '@testing-library/react';
export { customRender as render };
