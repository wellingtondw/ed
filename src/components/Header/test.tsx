import { render } from 'utils/test-utils';
import { Header } from '.';

describe('<Header />', () => {
  it('Should be able to render correctly', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
