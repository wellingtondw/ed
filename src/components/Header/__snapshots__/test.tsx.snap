// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should be able to render correctly 1`] = `
<div>
  <div
    class="sc-gKXOVf dLgbJC"
  >
    <div
      class="sc-iBkjds SHzFk"
    >
      <div
        class="sc-gsnTZi fXVKGc"
      >
        <div
          class="sc-jSMfEi dUBfmM"
        >
          <div
            class="sc-dkzDqf gcYGAC"
          >
            <div
              class="sc-eCYdqJ la-DYnh"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <input
              class="sc-hKMtZM gJnIHa"
              placeholder="Pesquisar por filme"
              type="text"
              value=""
            />
          </div>
        </div>
        <ul
          aria-label="search results"
          class="sc-ftvSup fOpNgm"
        >
          <p>
            Nenhum resultado para a pesquisa
          </p>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
